src/main/java/uk/gov/hmcts/ccd/domain/service/common/AttributeBasedAccessControlService.java

```diff
@@ -0,0 +1,278 @@
+                                                  List<CaseEventDefinition> caseEventDefinitions,
+                                                  Set<String> userRoles,
+                                                  Predicate<AccessControlList> criteria) {
+        return super.canAccessCaseEventWithCriteria(eventId,
+            caseEventDefinitions,
+            userRoles,
+            criteria);
+    }
+
+    @Override
+    public boolean canAccessCaseFieldsWithCriteria(JsonNode caseFields,
+                                                   List<CaseFieldDefinition> caseFieldDefinitions,
+                                                   Set<String> userRoles,
+                                                   Predicate<AccessControlList> criteria) {
+        applyAccessProfileRules(caseFieldDefinitions);
+
+        return super.canAccessCaseFieldsWithCriteria(caseFields,
+            caseFieldDefinitions,
+            userRoles,
+            criteria);
+    }
+
+    @Override
+    public boolean canAccessCaseViewFieldWithCriteria(CommonField caseViewField,
+                                                      Set<String> userRoles,
+                                                      Predicate<AccessControlList> criteria) {
+        return super.canAccessCaseViewFieldWithCriteria(caseViewField,
+            userRoles,
+            criteria);
+    }
+
+    @Override
+    public boolean canAccessCaseFieldsForUpsert(JsonNode newData,
+                                                JsonNode existingData,
+                                                List<CaseFieldDefinition> caseFieldDefinitions,
+                                                Set<String> userRoles) {
+        applyAccessProfileRules(caseFieldDefinitions);
+        return super.canAccessCaseFieldsForUpsert(newData,
+            existingData,
+            caseFieldDefinitions,
+            userRoles);
+    }
+
+    @Override
+    public CaseUpdateViewEvent setReadOnlyOnCaseViewFieldsIfNoAccess(CaseUpdateViewEvent caseEventTrigger,
+                                                                     List<CaseFieldDefinition> caseFieldDefinitions,
+                                                                     Set<String> userRoles,
+                                                                     Predicate<AccessControlList> access) {
+        applyAccessProfileRules(caseFieldDefinitions);
+
+        return super.setReadOnlyOnCaseViewFieldsIfNoAccess(caseEventTrigger,
+            caseFieldDefinitions,
+            userRoles,
+            access);
+    }
+
+    @Override
+    public CaseUpdateViewEvent updateCollectionDisplayContextParameterByAccess(CaseUpdateViewEvent caseEventTrigger,
+                                                                               Set<String> userRoles) {
+        return super.updateCollectionDisplayContextParameterByAccess(caseEventTrigger,
+            userRoles);
+    }
```

Comment:
We're not doing anything different here so no need to override the method?
```suggestion
```

Reply:
removed